### PR TITLE
PROD-1247 updating react-date-time-group to react v14

### PR DESCRIPTION
**Pull request by reecemillsom**
#### What does this PR do? (please provide any background)
This PR updates the react-date-time-group to version 0.14.2 of react.

#### What tests does this PR have?
Unit tests, run npm test and make sure they all still pass.

#### How can this be tested?
Run npm start and open the example.js in the browser.
file:///pathTo/example.html

#### Screenshots / Screencast
http://grab.by/NGtY

#### What gif best describes how you feel about this work?

---

- [x] I have checked our general [contributing document](https://github.com/holidayextras/culture/blob/master/CONTRIBUTING.md) and the project specific [contributing document](../blob/master/CONTRIBUTING.md) (if present) and I'm happy for this to be reviewed.

#### Reviewers

**Review 1**
- [x] :+1:

**Review 2** \*
- [x] :+1:

**Review 3** _(optional)_
- [ ] :+1:

By adding a +1 you are confirming you have...
- Witnessed the work behaving as expected (this could be on the author's machine or screencast).
- Checked for coding anti-patterns.
- Checked for appropriate test coverage.
- Checked all the tests are passing.

\*  for HX this review must be completed by an SE, SA or Project Guru

### CODE REVIEW

**Comment by michaelcarter** [comment]
Taking :one:

**Comment by jameshhx** [comment]
Taking :one:

File: package.json [hunk 1]
```diff
@@ -23,23 +25,30 @@
   },
   "dependencies": {
     "moment": "^2.10.6",
-    "react": "^0.13.3",
-    "react-bootstrap": "^0.26.1",
-    "react-datepicker": "^0.13.0",
-    "react-time-select": "^1.0.2"
+    "react": "^0.14.2",
```
Comment (jameshhx) [inline package.json]: Module versions should really be locked down, unless you believe there's a reason not to? We lock down version numbers to ensure that if a newer version of a module were to be released, we can be sure it would not affect our modules. Not everyone respected subversion backwards compatibility, unfortunately. @shackpank ?

File: .travis.yml [hunk 2]
```diff
@@ -1,11 +1,10 @@
 language: node_js
 sudo: false
 install:
-  - npm install npm@2
-  - ./node_modules/.bin/npm install
+  - npm install -g npm@2
```
Comment (jameshhx) [inline .travis.yml]: Is there a reason we need a global install of `npm` here? Is this related to module caching? Could the global installation interfere with any other builds on Travis?

File: doc/example.js [hunk 3]
```diff
@@ -60,7 +61,7 @@ fns.render = function() {
       <DateTimeGroup value={myDate} onChange={fns.changeDate} />
 
       <h1>Localization (Incomplete)</h1>
-      <DateTimeGroup onChange={fns.changeDate} locales={['en-US']} />
```
Comment (michaelcarter) [inline doc/example.js]: As a English company, do we not use `en-GB`?

**Comment by michaelcarter** [comment]
Appreciate the effort on moving stuff over to enzyme! One point from me and a couple of valid queries from James.

File: doc/example.js (hunk 3 above)
Comment (reecemillsom) [inline doc/example.js]: I think this was there just to prove that we can localize to there time format

File: .travis.yml (hunk 2 above)
Comment (reecemillsom) [inline .travis.yml]: There was an issue with npm not installing my modules properly and couldnt find the react-time-select. I think it was caused from travis caching the modules and the install not working properly

File: package.json (hunk 1 above)
Comment (jameshhx) [inline package.json]: After speaking with @shackpank, I'm happy for module versions to update within minor versions. The thinking is that this is client side. If we were to lock down version numbers, then the client could potentially end up with numerous different versions of the same module.

**Comment by jameshhx** [comment]
:+1: from me - my points have been addressed. Good job :smile:

**Comment by reecemillsom** [comment]
Thanks @jameshhx

File: doc/example.js (hunk 3 above)
Comment (michaelcarter) [inline doc/example.js]: :+1:

**Comment by michaelcarter** [comment]
+1
